util/data_set_helpers.py

```diff
@@ -10,184 +9,131 @@
+
+        self.gpu_feeders = [GpuFeeder(self, i) for i in range(self.gpu_feeder_count)]
+
+    def start_queue_threads(self, session, coord):
+        batch_threads = []
+        for gpu_feeder in self.gpu_feeders:
+            batch_threads += gpu_feeder.start_queue_threads(session, coord)
+        return batch_threads
+
+    def close_queues(self, session):
+        for gpu_feeder in self.gpu_feeders:
+            gpu_feeder.close_queues(session)
+
+    def set_data_set(self, feed_dict, data_set):
+        index = self.sets.index(data_set)
+        assert index >= 0
+        feed_dict[self.ph_queue_selector] = index
+        feed_dict[self.ph_batch_size] = data_set.batch_size
+
+    def next_batch(self, gpu_feeder_index):
+        return self.gpu_feeders[gpu_feeder_index].next_batch()
 
 class DataSet(object):
-    def __init__(self, files_list, thread_count, batch_size, numcep, numcontext, next_index=lambda x: x + 1):
-        self._coord = None
-        self._numcep = numcep
-        self._x = tf.placeholder(tf.float32, [None, numcep + (2 * numcep * numcontext)])
-        self._x_length = tf.placeholder(tf.int32, [])
-        self._y = tf.placeholder(tf.int32, [None,])
-        self._y_length = tf.placeholder(tf.int32, [])
-        self.example_queue = tf.PaddingFIFOQueue(shapes=[[None, numcep + (2 * numcep * numcontext)], [], [None,], []],
-                                                  dtypes=[tf.float32, tf.int32, tf.int32, tf.int32],
-                                                  capacity=2 * self._get_device_count() * batch_size)
-        self._enqueue_op = self.example_queue.enqueue([self._x, self._x_length, self._y, self._y_length])
-        self._close_op = self.example_queue.close(cancel_pending_enqueues=True)
+    def __init__(self, csvs, batch_size, skip=0, limit=0, ascending=True, next_index=lambda i: i + 1):
         self.batch_size = batch_size
-        self._numcontext = numcontext
-        self._thread_count = thread_count
-        self._files_list = self._create_files_list(files_list)
-        self._next_index = next_index
-
-    def _get_device_count(self):
-        available_gpus = get_available_gpus()
-        return max(len(available_gpus), 1)
+        self.next_index = next_index
+        self.files = None
+        for csv in csvs:
+            file = pandas.read_csv(csv)
+            if self.files is None:
+                self.files = file
+            else:
+                self.files = files.append(file)
+        self.files = self.files.sort_values(by="wav_filesize", ascending=ascending) \
+                         .ix[:, ["wav_filename", "transcript"]] \
+                         .values[skip:]
+        if limit > 0:
+            self.files = self.files[:limit]
+        self.total_batches = int(ceil(len(self.files) / batch_size))
+
+class DataSetQueue(object):
+    def __init__(self, data_sets, data_set):
+        self.data_sets = data_sets
+        self.data_set = data_set
```

Comment:
These two are never used outside of `DataSetQueue` and should have a leading underscore.